src/testing/benchmark.go

```diff
@@ -181,14 +181,23 @@ func (b *B) launch() {
 func (b *B) Run(name string, f func(b *B)) bool {
 	b.hasSub = true
 	sub := &B{
-		common:    common{name: name},
+		common: common{
+			name:     name,
+			finished: make(chan struct{}),
+		},
 		benchFunc: f,
 		benchTime: b.benchTime,
 	}
-	if sub.run1() {
-		sub.run()
-	}
-	b.add(sub.result)
+	go func() {
```

Comment:
I very intentionally avoided running tests in a goroutine, to reduce memory consumption. In fact, this was the main reason why I decided to reimplement the testing package instead of reusing the standard library version.
I think we should continue to avoid goroutines in the testing package.